src/split_single_dyn_dim.cpp

```diff
@@ -36,36 +36,79 @@ inline namespace MIGRAPHX_INLINE_NS {
 struct dynamic_dimensions_check
 {
     std::string dyn_param_str;
-    size_t dyn_index;
-    size_t min_dim;
-    size_t max_dim;
+    shape::dynamic_dimension dd;
 };
 
-optional<dynamic_dimensions_check>
-has_one_dyn_dim(const std::unordered_map<std::string, shape>& param_shapes)
+/**
+ * Returns value if the parameters contain one non-fixed dynamic_dimension that is the same between
+ * all of the dynamic shape parameters.
+ * Returns the parameters and the dynamic dimension in a vector of dynamic_dimensions_check objects.
+ */
+optional<std::vector<dynamic_dimensions_check>>
+has_one_unique_dyn_dim(const std::unordered_map<std::string, shape>& param_shapes)
 {
-    // True if parameters contain exactly one dynamic shape with exactly one non-fixed
-    // dynamic_dimension.
+    if(param_shapes.empty())
+    {
+        return std::nullopt;
+    }
     auto is_dynamic = [](const auto& p) { return p.second.dynamic(); };
-    auto ps_it      = std::find_if(param_shapes.begin(), param_shapes.end(), is_dynamic);
-    if(ps_it == param_shapes.end())
+    std::vector<std::decay_t<decltype(param_shapes)>::value_type> dyn_params{};
+    std::copy_if(
+        param_shapes.begin(), param_shapes.end(), std::back_inserter(dyn_params), is_dynamic);
+    if(dyn_params.empty())
         return std::nullopt;
-    // Check if there is a second dynamic parameter
-    if(std::any_of(std::next(ps_it), param_shapes.end(), is_dynamic))
+    std::vector<dynamic_dimensions_check> ret{};
+    // get non-fixed dynamic_dimension from all parameters
+    for(const auto& param : dyn_params)
+    {
+        const auto& dds   = param.second.dyn_dims();
+        int num_non_fixed = 0;
+        for(const auto& dd : dds)
+        {
+            if(not dd.is_fixed())
+            {
+                num_non_fixed += 1;
+                // catch more than one non-fixed dynamic_dimension
+                if(num_non_fixed > 1)
+                {
+                    return std::nullopt;
+                }
+                ret.push_back(dynamic_dimensions_check{param.first, dd});
```

Comment:
This can technically return true if there are more than one non-fixed dynamic dimensions in all parameters with same shape. If so docstring need to be modified.

Reply:
I think you're saying that the docstring should be updated to make it more clear that the same non-fixed `dynamic_dimension` can appear within multiple parameters?